paddle/phi/kernels/onednn/matmul_kernel.cc

```diff
@@ -0,0 +1,450 @@
+    PADDLE_ENFORCE_EQ(
+        (std::is_same<YT, float>::value),
+        true,
+        errors::InvalidArgument(
+            "Input(Y) must be fp32 data type since only fp32 data type is "
+            "supported in the current design of MKLDNN INT8."));
+
+    auto x_matrix = UpdateDataFormat<XT>(x_input, x_num_col_dims, ctx);
+    auto y_matrix = UpdateDataFormat<YT>(y_input, y_num_col_dims, ctx);
+
+    auto output_dim = output->dims();
+    if (output_dim.size() != 2) {
+      output->Resize({x_matrix.dims()[0], y_matrix.dims()[1]});
+    }
+
+    if (mul_) {
+      UpdateDataPointers(ctx, output, &x_matrix);
+      Execute();
+      return *(mul_);
+    }
+
+    auto src_desc =
+        CreateMemDescriptor<XT>(&x_matrix, funcs::OneDNNMemoryFormat::nc);
+    x_input_ = CreateMemory<XT>(src_desc, &x_matrix);
+
+    if (is_int8_) {
+      const auto trans_y = TransposeInputY(&y_matrix);
+      auto scale_y =
+          ctx.HasDnnAttr("scale_y")
+              ? PADDLE_GET_CONST(std::vector<float>, ctx.GetDnnAttr("scale_y"))
+              : std::vector<float>();
+      y_input_ = QuantInputY(trans_y, scale_y);
+    } else {
+      y_input_ = TransposeInputY(&y_matrix);
+    }
+
+    auto dst_desc =
+        CreateMemDescriptor<OT>(output, funcs::OneDNNMemoryFormat::any);
+
+    mul_ = CreateMulPrimitive(*x_input_, *y_input_, dst_desc, output, ctx);
+    Execute();
+    return *(mul_);
+  }
+
+ private:
+  memory ReorderWithScale(const memory::desc &src_desc,
+                          const memory::desc &dst_desc,
+                          void *src_data,
+                          const std::vector<float> &scale) {
+    auto mask = scale.size() > 1 ? 1 : 0;
+    dnnl::primitive_attr attr;
+    attr.set_output_scales(mask, scale);
+
+    auto src_mem = memory(src_desc, engine_, src_data);
+    auto dst_mem = memory(dst_desc, engine_);
+
+    auto reorder_pd = dnnl::reorder::primitive_desc(src_mem, dst_mem, attr);
+
+    auto reorder = dnnl::reorder(reorder_pd);
+
+    auto &astream = OneDNNContext::tls().get_stream();
+    {
+      paddle::platform::RecordEvent record_reorder(
+          "int_reorder",
+          paddle::platform::TracerEventType::UserDefined,
+          2,
+          paddle::platform::EventRole::kUniqueOp);
+      reorder.execute(astream, src_mem, dst_mem);
+      astream.wait();
+    }
+
+    return dst_mem;
+  }
+
+  memory QuantInputY(memory input_y, const std::vector<float> &scale_y) {
+    const auto &dims = input_y.get_desc().data.dims;
+    auto ndims = input_y.get_desc().data.ndims;
+    auto y_dims = std::vector<int64_t>(dims, dims + ndims);
+
+    auto user_y_desc =
+        CreateMemDescriptor<YT>(y_dims, funcs::OneDNNMemoryFormat::oi);
+    auto y_desc =
+        CreateMemDescriptor<int8_t>(y_dims, funcs::OneDNNMemoryFormat::oi);
+
+    return ReorderWithScale(
+        user_y_desc, y_desc, input_y.get_data_handle(), scale_y);
+  }
+
+  dnnl::primitive_attr CreateMulAttr(const OneDNNContext &ctx,
```

Comment:
```suggestion
  dnnl::primitive_attr CreateMulAttr(const OneDNNContext &ctx,
```
Please rename each OneDNNContext `ctx` to `dev_ctx` to be consistent with already migrated oneDNN kernels